File: Programmers/Lv_1/덧칠하기.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int solution(int n, int m, vector<int> section) {
    int answer = 0;
    int cnt = 0; 
    int current = 0;

    for (int i = 0; i < section.size(); i++) {
        if (current > section[i]) continue;
        cnt++;
        current = section[i] + m;
    }
    answer = cnt;
    return answer;
}

int solution2(int n, int m, vector<int> section) {
    int answer = 1;

    int pivot = section[0];
    for (const auto v: section) {
        if (v < pivot + m) continue;
        else {
            pivot = v;
            answer++;
        }
    }
    return answer;
}

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int n, m, num, data;
    vector<int> section;
    cin >> n >> m;
    cin >> num;
    for (int i = 0; i < num; i++) {
        cin >> data;
        section.push_back(data);
    }

    cout << solution(n, m, section);

    return 0;
}

File: Programmers/Lv_1/덧칠하기.py
def solution(n, m, section):
    ans, cnt, cur = 0, 0, 0
    for i in range(len(section)):
        if cur > section[i]:
            continue
        cnt += 1
        cur = section[i] + m
        
    ans = cnt
    return ans

def solution2(n, m, section):
    ans = 1
    prev = section[0]
    
    for sec in section:
        if sec-prev >= m:
            prev = sec
            answer += 1
            
    return ans
            